### PR TITLE
Windows Portable App

### PR DESCRIPTION
**Pull request by feross**
Fixes https://github.com/feross/webtorrent-desktop/issues/358

### Background

- [What is a portable app?](http://portableapps.com/about/what_is_a_portable_app)
- [Portable Audacity](http://wiki.audacityteam.org/wiki/Portable_Audacity)

### Changes

- A new `*-win.zip` output file is now produced, which just contains the electron-packaged windows app, with an added empty "Portable Settings" folder.
- At startup, when a "Portable Settings" folder is found next to the WebTorrent.exe file, then the app is in "Portable Mode" which entails:
  - Electron "userData" folder is set to the "Portable Settings" folder
  - The `config.json` file is saved in the "Portable Settings" folder
  - Torrent posters and torrent files are saved in the "Portable Settings" folder
  - The default download folder is set to the "Portable Settings" folder

Am I forgetting anything?

### PortableApps.com Platform

I [started a discussion](http://portableapps.com/node/54146) to get WebTorrent Desktop included in the PortableApps.com bundle.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,8 +57,16 @@ Where `[platform]` is `darwin`, `linux`, `win32`, or `all` (default).
 
 The following optional arguments are available:
 
-- `--package=[type]` - Package only one output file. `type` is `deb`, `dmg`, `zip`, or `all` (default)
 - `--sign` - Sign the application (OS X, Windows)
+- `--package=[type]` - Package single output type.
```
Comment (dcposch) [inline README.md]: Looks good!

File: bin/package.js [hunk 2]
```diff
@@ -348,65 +346,90 @@ function buildWin32 (cb) {
...
+        console.log('Windows: Created installer.')
+        cb(null)
+      }).catch(cb)
+    }
+
+    function packagePortable (cb) {
+      // Create Windows portable app
+      console.log('Windows: Creating portable app...')
+
+      var portablePath = path.join(buildPath[0], 'Portable Settings')
```
Comment (dcposch) [inline bin/package.js]: So this just adds an empty Portable Settings folder to the build path and then zips it. Simpler than above -- we're not making a self extracting exe. Seems elegant

File: bin/package.js [hunk 3]
```diff
@@ -348,65 +346,90 @@ function buildWin32 (cb) {
...
 }
 
 function buildLinux (cb) {
-  var distPath = path.join(config.ROOT_PATH, 'dist')
-
   console.log('Linux: Packaging electron...')
   electronPackager(Object.assign({}, all, linux), function (err, buildPath) {
     if (err) return cb(err)
-    console.log('Linux: Packaged electron.')
+    console.log('Linux: Packaged electron. ' + buildPath[0])
```
Comment (dcposch) [inline bin/package.js]: What was wrong with the Linux build? Does `packageDeb` need to finish before `packageZip` can start?

File: package.json [hunk 4]
```diff
@@ -15,8 +15,7 @@
   },
   "dependencies": {
     "airplay-js": "guerrerocarlos/node-airplay-js",
-    "application-config": "^0.2.0",
-    "application-config-path": "^0.1.0",
+    "application-config": "feross/node-application-config",
```
Comment (dcposch) [inline package.json]: +1

The original was janky.

If you've already made your own module, you can also use `JSON.stringify(obj, null '\t')` to produce neatly printed configuration files. (Our current config.jsons are all on one line.)

**Comment by dcposch** [comment]
LGTM

File: package.json (hunk 4 above)
Comment (feross) [inline package.json]: Good idea.

File: bin/package.js (hunk 3 above)
Comment (feross) [inline bin/package.js]: The change is so that `cb` is always called. It was sometimes not called before.

File: bin/package.js (hunk 2 above)
Comment (feross) [inline bin/package.js]: Yep!

**Pull request merged**